src/rascal/models/sparse_kernel_predict.hh

```diff
@@ -0,0 +1,256 @@
+/**
+ * @file   rascal/models/sparse_kernel_predict.hh
+ *
+ * @author Felix Musil <felix.musil@epfl.ch>
+ *
+ * @date   15 Sept 2020
+ *
+ * @brief Implementation of similarity kernels for sparse methods
+ *
+ * Copyright 2020 Felix Musil COSMO (EPFL), LAMMM (EPFL)
+ *
+ * Rascal is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation, either version 3, or (at
+ * your option) any later version.
+ *
+ * Rascal is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this software; see the file LICENSE. If not, write to the
+ * Free Software Foundation, Inc., 59 Temple Place - Suite 330,
+ * Boston, MA 02111-1307, USA.
+ */
+
+#ifndef SRC_RASCAL_MODELS_SPARSE_KERNEL_PREDICT_HH_
+#define SRC_RASCAL_MODELS_SPARSE_KERNEL_PREDICT_HH_
+
+#include "rascal/math/utils.hh"
+#include "rascal/models/kernels.hh"
+#include "rascal/models/sparse_kernels.hh"
+#include "rascal/structure_managers/property_block_sparse.hh"
+#include "rascal/structure_managers/structure_manager_collection.hh"
+#include "rascal/utils/json_io.hh"
+
+namespace rascal {
+
+  /**
+   * Compute the partial gradients of a structure w.r.t atomic positions
+   * using the GAP model (see
+   * @ref SparseKernelImpl<internal::SparseKernelType::GAP>::compute_derivative)
+   *
+   * The point of this function is to provide a faster prediction routine
+   * compared to computing the kernel element and then multiplying them with
+   * the weights of the model.
```

Comment:
Perhaps mention here why the latter method is slower? How does it scale compared to this function?